Lib/pdb.py

```diff
@@ -597,11 +597,20 @@ def precmd(self, line):
         args = line.split()
         while args[0] in self.aliases:
             line = self.aliases[args[0]]
-            ii = 1
-            for tmpArg in args[1:]:
-                line = line.replace("%" + str(ii),
-                                      tmpArg)
-                ii += 1
+            for idx in range(1, 10):
+                if f'%{idx}' in line:
+                    if idx >= len(args):
+                        self.error(f"Not enough arguments for alias '{args[0]}'")
+                        # This is a no-op
+                        return "!"
+                    line = line.replace(f'%{idx}', args[idx])
+                elif '%*' not in line:
+                    if idx < len(args):
+                        self.error(f"Too many arguments for alias '{args[0]}'")
+                        # This is a no-op
+                        return "!"
```

Comment:
Why is this return needed?

Reply:
Because `onecmd` expects a command, and returning a `False`-like value would be equivalent to returning an empty line - and cmd considers an empty like as "repeat last command". If you do "enter" in pdb, that's repeat last command.